### ISSUE TITLE
Support popular frontend javascript frameworks

### ISSUE THREAD

**Issue by kevaundray** [noir-lang/noir#2007]
### Problem

Many times users are using a frontend framework and want to integrate our tools into their framework.

As it stands, some frameworks are not usable with our tools.

One of these is vitejs, see [here](https://github.com/noir-lang/noir/issues/1075)

### Happy Case

Its a bit of a dying battle, but it would be good to support all of the popular frameworks or keep track of the popular frameworks that a user may want to work with our setup.

It may be that if we have X number of frameworks then we just need to keep those updated.

**Note**

This issue can be split into two:

- Ensuring compatibility for compiler packages
- Ensuring compatibility for barretenberg.js

### Additional Context

This may not be the right place to put this issue, but lets keep it here and when we figure out a better place to put it then we can move.

### Would you like to submit a PR for this Issue?

No

**Comment by signorecello** [comment]
Looks great, though currently blocked by the existing packages. c.c. @Savio-Sou so we can have this on the roadmap

**Comment by Savio-Sou** [comment]
Strategically the core team will maintain the packages with Next.js compatibility to ensure Noir has at least one go-to frontend framework while reserving engineering bandwidth for other initiatives in the short- to mid-term. The core team might revisit this in the longer term but we have no concrete plans atm.

If anyone is interested in adding support for additional frontend frameworks, all is welcome to create Issues + PRs for frameworks of interest. The core team would be more than happy to provide support accordingly 🙌

**Comment by kevaundray** [comment]
I don't think we currently exclude any frontends -- marking this as completed unless we can find a specific popular framework that we do not support. In that case, we can open up a separate issue in either bb.js or this repository

**Comment by signorecello** [comment]
We should be able to tackle everything that comes to a web browser now 💪
